File: src/Entity/Rapport.php
<?php

namespace App\Entity;
use Symfony\Component\Validator\Constraints as Assert;
use App\Repository\RapportRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: RapportRepository::class)]
class Rapport
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]



    private ?int $id = null;



     /**
     * @Assert\NotBlank(message="La note ne peut pas être vide.")
     * @Assert\Regex(
     *     pattern="/^[a-zA-Z\s]+$/",
     *     message="La note ne doit contenir que des lettres et des espaces."
     * )
     */
    #[ORM\Column(length: 255)]
    private ?string $note = null;


     /**
     * @Assert\NotBlank(message="Le type ne peut pas être vide.")
     * @Assert\Choice(
     *     choices={"diabetique", "consultation_psychologique", "resultat_analyses"},
     *     message="Le type doit être l'un des choix proposés."
     * )
     */
    #[ORM\Column(length: 255)]
    private ?string $type = null;

    #[ORM\OneToOne(cascade: ['persist', 'remove'])]
    private ?Rendezvous $rdv = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNote(): ?string
    {
        return $this->note;
    }

    public function setNote(string $note): static
    {
        $this->note = $note;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): static
    {
        $this->type = $type;

        return $this;
    }

    public function getRdv(): ?Rendezvous
    {
        return $this->rdv;
    }

    public function setRdv(?Rendezvous $rdv): static
    {
        $this->rdv = $rdv;

        return $this;
    }
}

File: src/main/java/tn/esprit/applicationgui/Rapport.java
package tn.esprit.applicationgui;

public class Rapport {
    private String note;
    private String type;
    private int rdvId;

    public Rapport(String note, String type, int rdvId) {
        this.note = note;
        this.type = type;
        this.rdvId = rdvId;
    }
    public Rapport(int id, String note, String type, int rdvId) {

        this.note = note;
        this.type = type;
        this.rdvId = rdvId;
    }


    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note = note;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public int getRdvId() {
        return rdvId;
    }

    public void setRdvId(int rdvId) {
        this.rdvId = rdvId;
    }
}
